spec/controllers/books_controller_spec.rb

```diff
@@ -1,4 +1,43 @@
 require 'rails_helper'
+require 'support/shared_context'
 
 RSpec.describe BooksController, type: :controller do
+  include_context 'Authenticated User'
```

Comment:
What if we want to test and endpoint with an unauthenticated user? Let's refactor this a little!
With an structure similar to this 👇 , we can test each endpoint with authenticated and unauthenticated requests
```ruby
describe BooksController do
  describe 'GET #index' do
    context 'with an authenticated user' do
      include_context 'Authenticated User'
      ....
    end

    context 'without an authenticated user' do
      ....
    end
  end
end
```